packages/SwingSet/src/kernel/state/vatKeeper.js

```diff
@@ -148,8 +152,14 @@ export function makeVatKeeper(
     const kernelKey = `${vatID}.c.${kernelSlot}`;
     const vatKey = `${vatID}.c.${vatSlot}`;
     kdebug(`Delete mapping ${kernelKey}<=>${vatKey}`);
-    storage.delete(kernelKey);
-    storage.delete(vatKey);
+    if (storage.has(kernelKey)) {
```

Comment:
are there situations where this is false?

Reply:
One further inspection, probably not, but suspenders and belt...